### PR TITLE
opencv3: fix missing /usr/bin/python in macOS 12.3 and enable python 3.10 variant

### PR DESCRIPTION
**Pull request by jsalort**
#### Description

Fix compilation of opencv3 under macOS 12.3 by changing the default Python from `/usr/bin/python` to `/usr/bin/python3`. My only concern is that I don't know if earlier macOS had /usr/bin/python3. If they don't, then tests should be added to choose the default python accordingly...

I also enabled python 3.10 variant.

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 12.3.1 21E258 arm64
Xcode 13.3.1 13E500a

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

Note: No existing ticket. No existing tests.

### CODE REVIEW

File: graphics/opencv3/Portfile [hunk 1]
```diff
@@ -379,7 +379,7 @@ foreach python_branch {2.7} {
     }
 }
 
-set python_branches {3.5 3.6 3.7 3.8 3.9}
+set python_branches {3.5 3.6 3.7 3.8 3.9 3.10}
```
Comment (reneeotten) [inline graphics/opencv3/Portfile:382]: let's remove the EOL Pythin branches `3.5` and `3.6`

**Comment by Gcenx** [comment]
> at least Catalina also has `/usr/bin/python3`, but why not just depend on the default Python version from MacPorts (currently 3.10) to avoid using a system version?

Agreed no point using the system provided python(3)

**Comment by jsalort** [comment]
> > at least Catalina also has `/usr/bin/python3`, but why not just depend on the default Python version from MacPorts (currently 3.10) to avoid using a system version?
> 
> Agreed no point using the system provided python(3)

Then what port should it depend on? python310? Or is there a way to depend on the "default" python?

File: graphics/opencv3/Portfile (hunk 1 above)
Comment (jsalort) [reply graphics/opencv3/Portfile]: OK.
Question about the "Request feature" of Github: should I click on the "Resolve conversation" button, or do you?

**Comment by ryandesign** [comment]
/usr/bin/python3 did not exist prior to macOS 10.15.

Lint says:
> ```
> Warning: Dependency port:python310 specified multiple times in depends_lib
> ```

I guess that applies to the py310-opencv3 subport. The GitHub web interface doesn't give me the ability to make suggestions on lines that you did not edit, but probably the fix is to change this part of the python subports section:

```
        depends_lib-append \
                            port:${name} \
                            port:python${python_version} \
                            port:py${python_version}-numpy
```

to this:

```
        depends_lib-append \
                            port:${name} \
                            port:py${python_version}-numpy
        depends_lib-replace \
                            port:python310 \
                            port:python${python_version}
```

"310" and "3.10" is getting repeated in an awful lot of places in this new portfile. It would be better to define "3.10" in a single variable that can be referenced wherever needed. That way, every January when MacPorts promotes a new major version of Python to be the default, only a single-line change will be needed to this portfile.

**Comment by mascguy** [comment]
Looks good at this point. Currently testing locally with some variants enabled - including `contrib` - to be sure all is well.

Assuming there aren't any issues, I'll merge.

Of note, I'll also apply similar changes to `opencv4`, along with `opencv3-devel` and `opencv4-devel`.

**Comment by mascguy** [comment]
And thanks so much for your work on this Julien!

**Review by mascguy — APPROVED** [review]

**Pull request merged**